pkg/callbackserver/callbackserver.go

```diff
@@ -72,16 +72,16 @@ func (cs *CallbackServer) Run(ctx context.Context, callback CallbackFunc) (inter
 			Handler: mux,
 		}
 	}
+	defer server.Shutdown(ctx)
 
-	// Start the server.
-	go (func(s *http.Server) {
```

Comment:
Removed the function argument, as the scope of the goroutine will not outlive the scope of the parent function, there's no need to copy the reference